src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -376,32 +367,10 @@ internal static bool IsValidHostObjectType(Type type)
         /// <paramref name="hasValue"/> is false in the former case and true
         /// in the latter.
         /// </remarks>
-        public ITypeSymbol GetSubmissionResultType(out bool hasValue)
-        {
-            return CommonGetSubmissionResultType(out hasValue);
-        }
-
-        protected abstract ITypeSymbol CommonGetSubmissionResultType(out bool hasValue);
+        internal abstract ITypeSymbol GetSubmissionResultType(out bool hasValue);
```

Comment:
The usual .net pattern is to call the method `TryGetSomething`, return `bool`, and have the `Something` be an out parameter.